modules/miner/miner.go

```diff
@@ -194,10 +199,15 @@ func New(cs modules.ConsensusSet, tpool modules.TransactionPool, w modules.Walle
 // Close terminates all ongoing processes involving the miner, enabling garbage
 // collection.
 func (m *Miner) Close() error {
+	if err := m.tg.Stop(); err != nil {
```

Comment:
I think the latter is the correct approach. Threads should use `tg` as a point of synchronization for knowing when to shut down.